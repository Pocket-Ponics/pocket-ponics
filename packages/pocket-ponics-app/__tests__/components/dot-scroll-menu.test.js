import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'

import DotScrollMenu from '../../components/greenhouse/dot-scroll-menu'

describe('Greenhouse Display', () => {
	test('renders correctly', () => {
		const tree = shallow(
			<DotScrollMenu 
				greenhouseList={[{},{},{}]}
				current={0}/>).html()
		expect(tree).toMatchSnapshot()
	})

	test('renders with last selected', () => {
		const tree = shallow(
			<DotScrollMenu 
				greenhouseList={[{},{},{}]}
				current={2}/>).html()
		expect(tree).toMatchSnapshot()
	})

	test('renders with only 1 item', () => {
		const tree = shallow(
			<DotScrollMenu 
				greenhouseList={[{}]}
				current={0}/>).html()
		expect(tree).toMatchSnapshot()
	})

	test('clicks on buttons', () => {
		const swapItem = jest.fn()
		const tree = shallow(
			<DotScrollMenu 
				greenhouseList={[{},{}]}
				current={0}
				swapItem={swapItem}/>)

		tree.find(TouchableOpacity).at(0).props().onPress()
		tree.find(TouchableOpacity).at(1).props().onPress()

		expect(swapItem).toHaveBeenCalledTimes(2)
	})
})