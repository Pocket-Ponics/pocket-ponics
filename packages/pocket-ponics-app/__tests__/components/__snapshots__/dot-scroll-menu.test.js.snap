// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greenhouse Display renders correctly 1`] = `"<View style=\\"align-content:center\\"><View style=\\"flex-direction:row;align-items:center;justify-content:center\\"><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF\\">•</Text></View><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF;opacity:0.7\\">•</Text></View><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF;opacity:0.7\\">+</Text></View></View></View>"`;

exports[`Greenhouse Display renders with last selected 1`] = `"<View style=\\"align-content:center\\"><View style=\\"flex-direction:row;align-items:center;justify-content:center\\"><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF;opacity:0.7\\">•</Text></View><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF;opacity:0.7\\">•</Text></View><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF\\">+</Text></View></View></View>"`;

exports[`Greenhouse Display renders with only 1 item 1`] = `null`;
